File: hang.liu/1185.js
var dayOfTheWeek = function(day, month, year) {
  const weekMap = [
    "Sunday",
    "Monday",
    "Tuesday",
    "Wednesday",
    "Thursday",
    "Friday",
    "Saturday"
  ];
  // return weekMap[new Date(year, month - 1, day).getDay()]

  var isLeap = year => (year % 4 === 0 && year % 100 !== 0) || year % 400 == 0;
  var months = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
  var days = 0;

  for (var i = 1971; i < year; i++) {
    days += 365;
    if (isLeap(i)) days += 1;
  }

  for (var i = 0; i < month - 1; i++) {
    days += months[i];
  }

  if (month > 2 && isLeap(year)) days += 1;

  days += day;

  return weekMap[(5 - 1 + days) % 7];
};

File: xiang.li/1185.kt
class Solution {
    fun dayOfTheWeek(day: Int, month: Int, year: Int): String {
        //蔡勒公式
        val week = arrayOf("Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday")
        val w = if (month <= 2) {
            val y = (year - 1) % 100
            val c = ((year - 1) - y) / 100
            ((y + y / 4 + c / 4 - 2 * c + 26 * (month + 12 + 1) / 10 + day - 1) % 7 + 7) % 7
        } else {
            val y = year % 100
            val c = (year - y) / 100
            ((y + y / 4 + c / 4 - 2 * c + 26 * (month + 1) / 10 + day - 1) % 7 + 7) % 7
        }
        return week[w]
    }

